packages/block-editor/src/components/rich-text/index.js

```diff
@@ -289,6 +299,24 @@ export function RichTextWrapper(
 		anchorRef.current?.focus();
 	}
 
+	const bindings = getBlockAttributes( clientId )?.metadata?.bindings;
+	const blockTypeAttributes = getBlockType( blockName ).attributes;
+	const { getBlockBindingsSource } = unlock( useSelect( blockEditorStore ) );
+	let shouldDisableEditing = false;
+	if ( bindings )
+		for ( const [ attribute, settings ] of Object.entries( bindings ) ) {
+			// If any of the attributes with source "rich-text" is part of the bindings,
+			// has a source with `lockAttributesEditing`, disable it.
+			if (
+				blockTypeAttributes?.[ attribute ]?.source === 'rich-text' &&
```

Comment:
This could also possibly lock RichText fields that aren't connected, if there is one that is connected and other that aren't. It isn't an issue for the blocks that have binding allowed, but I'm raising it as a future issue.

Reply:
Are there any blocks with two attributes connected to "rich-text"? If so, is there a way to know which attribute the RichText corresponds to from the RichText?

Reply:
Sure, Pullquote would be an example. It looks like the `identifier` props was used in the past whenever `RichText` needed to distinguish between wired attributes:

https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/pullquote/edit.js

@ellatrix, is it still supported? Have you ever considered moving the control for the block attribute instead of passing down the value to the `RichText` component living in the block editor? It would be convenient in this particular case.

Reply:
What do you mean with moving the control? Yes identifier is currently used almost everywhere, it's needed to store selection

Reply:
Instead of

```js
<RichText
	tagName="pre"
	identifier="content"
	preserveWhiteSpace
	value={ content }
	onChange={ ( nextContent ) => {
		setAttributes( {
			content: nextContent,
		} );
	} }
	...   
```

we could have:

```js
<RichText
	tagName="pre"
	attributeName="content"
	preserveWhiteSpace
	...   
```

so `value` and `onChange` could be automatically handled internally (when not provided) based on the `attributeName` (we could keep the `identifier` as name).

Reply:
Yes, I considered that, but the ship has kind of sailed. We can try it again if you like, I don't remember the obstacles.

Reply:
You can use identifier, but it's not guaranteed to exist :)

Reply:
We could always make it a requirement for block binding. At least for v1.

Reply:
The discussion continues in https://github.com/WordPress/gutenberg/issues/58233.